src/tox/session.py

```diff
@@ -564,47 +517,14 @@ def installpkg(self, venv, path):
                 venv.status = sys.exc_info()[1]
                 return False
 
-    def get_installpkg_path(self):
-        """
-        :return: Path to the distribution
-        :rtype: py.path.local
-        """
-        if not self.config.option.sdistonly and (
-            self.config.sdistsrc or self.config.option.installpkg
-        ):
-            path = self.config.option.installpkg
-            if not path:
-                path = self.config.sdistsrc
-            path = self._resolve_package(path)
-            self.report.info("using package {!r}, skipping 'sdist' activity ".format(str(path)))
-        else:
-            try:
-                path = self._makesdist()
-            except tox.exception.InvocationError:
-                v = sys.exc_info()[1]
-                self.report.error("FAIL could not package project - v = {!r}".format(v))
-                return
-            sdistfile = self.config.distshare.join(path.basename)
-            if sdistfile != path:
-                self.report.info("copying new sdistfile to {!r}".format(str(sdistfile)))
-                try:
-                    sdistfile.dirpath().ensure(dir=1)
-                except py.error.Error:
-                    self.report.warning(
-                        "could not copy distfile to {}".format(sdistfile.dirpath())
-                    )
-                else:
-                    path.copy(sdistfile)
-        return path
-
     def subcommand_test(self):
         if self.config.skipsdist:
             self.report.info("skipping sdist step")
-            path = None
         else:
-            path = self.get_installpkg_path()
-            if not path:
-                return 2
+            for venv in self.venvlist:
```

Comment:
We now allow separate packages per virtual environment. The use case for this extension is to allow e.g. binary packages (might need different compilation) and/or wheels (different wheel per venv).